core/domain/exp_domain.py

```diff
@@ -655,6 +659,75 @@ def validate(self):
         self.outcome.validate()
 
 
+class Hint(object):
+    """Value object representing a hint."""
+    def __init__(self, hint_text):
+        self.hint_text = html_cleaner.clean(hint_text)
+
+    def to_dict(self):
+        return {
+            'hint_text': self.hint_text,
+        }
+
+    @classmethod
+    def from_dict(cls, hint_dict):
+        return cls(html_cleaner.clean(hint_dict['hint_text']))
+
+    def validate(self):
+        if not isinstance(self.hint_text, basestring):
+            raise utils.ValidationError(
+                'Expected hint text to be a string, received %s' %
+                self.hint_text)
+
+
+class Solution(object):
+    """Value object representing a solution.
+
+    A solution consists of answer_is_exclusive, correct_answer and an
+    explanation.answer_is_exclusive when True indicates that it is the only
+    correct answer and False indicates that it is one possible answer.
+    correct_answer records the answer to progress to the next card and
+    explanation is a string containing an explanation for the solution.
+    """
+    def __init__(self, answer_is_exclusive, correct_answer, explanation):
+        self.answer_is_exclusive = answer_is_exclusive
+        self.correct_answer = (
+            html_cleaner.clean(correct_answer)
```

Comment:
No, that's not right. You need to actually look at the type of the answer and clean any subfields in it of type 'html'. It's possible that the normalizer functions in schema_utils can do this for you, but you'll need to check.